### PR TITLE
:bug: Fixes #28 Enable tar on agent

### LINKED ISSUE #28: Getting Error while running on AKS Pods
Hi,

I am trying to run this plugin on Azure Kubernetes Pods, while it works perfectly fine on master node but it failed on pods with below error 
`Using local directory /home/jenkins/workspace/acr_build_task/. as docker build source.
Compressing local directory into /home/jenkins/workspace/acr_build_task/./a0880edd-d46a-4a22-a9b1-9c2f444241fe.tar.gz.
Compressing ignore rules a0880edd-d46a-4a22-a9b1-9c2f444241fe.tar.gz.
ERROR: No such file or directory
ERROR: ACR Quick Task ended with HasError`

### PR DESCRIPTION
**Pull request by yuwzho**
Fixes #28

### CODE REVIEW

**Comment by yuwzho** [comment]
Download and try the [hpi file](https://github.com/yuwzho/azure-acr-plugin/releases/download/0.4.0-snapshot/azure-container-registry-tasks-plugin.hpi)

File: src/main/java/com/microsoft/jenkins/acr/commands/scm/LocalSCMCommand.java [hunk 1]
```diff
@@ -26,69 +30,94 @@
...
+            File file = new File(filename);
+            if (!file.exists()) {
+                return list;
+            }
 
-        String getRegistryName();
+            try {
+                BufferedLineReader reader = null;
+                try {
+                    reader = new BufferedLineReader(new InputStreamReader(new FileInputStream(file)));
```
Comment (gavinfish) [inline src/main/java/com/microsoft/jenkins/acr/commands/scm/LocalSCMCommand.java]: Use try-with-resources to skip unnecessary exceptions check and close().

File: src/main/java/com/microsoft/jenkins/acr/commands/scm/LocalSCMCommand.java [hunk 2]
```diff
@@ -26,69 +30,94 @@
...
+                       .compress()
+                       .fileList();
+                AzureStorageBlockBlob blob = new AzureStorageBlockBlob(url);
+                blob.uploadFile(tar.getRemote());
+                return filenames;
+            } catch (Exception e) {
+                throw e;
             } finally {
-                reader.close();
+                new File(tar.getRemote()).delete();
```
Comment (gavinfish) [inline src/main/java/com/microsoft/jenkins/acr/commands/scm/LocalSCMCommand.java]: Why not use FilePath#deleteRecursive()?

File: src/main/java/com/microsoft/jenkins/acr/commands/scm/LocalSCMCommand.java [hunk 3]
```diff
@@ -26,69 +30,94 @@
...
+               String[] filenames = CompressibleFileImpl.compressToFile(tar.getRemote())
+                       .withIgnoreList(ignoreList.toArray(new String[ignoreList.size()]))
+                       .withDirectory(Util.normalizeFilename(dir.getRemote()))
+                       .compress()
+                       .fileList();
+                AzureStorageBlockBlob blob = new AzureStorageBlockBlob(url);
+                blob.uploadFile(tar.getRemote());
+                return filenames;
+            } catch (Exception e) {
+                throw e;
```
Comment (gavinfish) [inline src/main/java/com/microsoft/jenkins/acr/commands/scm/LocalSCMCommand.java]: If do not add extra information to the exception, no need to have catch here.

File: src/main/java/com/microsoft/jenkins/acr/commands/scm/LocalSCMCommand.java [hunk 4]
```diff
@@ -26,69 +30,94 @@
...
-                    .fileList();
-            data.logStatus(
-                    Messages.scm_compress_files(StringUtils.join(filenames, Constants.LONG_LIST_SPERATE)));
+            String[] filenames = data.getJobContext().getWorkspace().act(agent);
+            data.logStatus(Messages.scm_compress_files(StringUtils.join(filenames, Constants.LONG_LIST_SPERATE)));
             data.logStatus(Messages.scm_upload(request.getUrl()));
-            AzureStorageBlockBlob blob = new AzureStorageBlockBlob(request.getUrl());
-            blob.uploadFile(localFileName);
         } catch (Exception e) {
+            data.logError(e);
```
Comment (gavinfish) [inline src/main/java/com/microsoft/jenkins/acr/commands/scm/LocalSCMCommand.java]: No need to log an exception and throw it again.

**Pull request merged**